amm-executor/src/Spectrum/Executor/Backlog/Data/BacklogOrder.hs

```diff
@@ -1,24 +1,27 @@
 module Spectrum.Executor.Backlog.Data.BacklogOrder
   ( BacklogOrder(..)
-  , WeightedOrder(..)
-  , mkWeightedOrder
+  , WeightedOrderWithTimestamp(..)
+  , mkWeightedOrderWithTimestamp
   ) where
 
 import Spectrum.Executor.Data.OrderState
   ( OrderState )
 import Spectrum.Executor.Types
   ( Order, OrderId, OrderWeight, weightOrder, orderId )
+import RIO.Time (UTCTime)
+import Data.Aeson.Types (ToJSON, FromJSON)
+import RIO (Generic)
 
 data BacklogOrder = BacklogOrder
-  { orderState   :: OrderState
-  , backlogOrder :: Order
-  }
+  { orderTimestamp :: UTCTime
```

Comment:
Do we really need this wrapper? We simply can keep it in seconds as long value

Reply:
Wrappers are good, timestamp can be represented in different units

Reply:
Not clear though, what exactly is UTCTime? Is it a unix time or more like a date-time ?

Reply:
it's just a wrapper around UTC, under the hood "data-time"